src/charm.py

```diff
@@ -261,12 +72,37 @@ def _on_collect_status(self, e: CollectStatusEvent):
     @property
     def tempo_endpoint(self) -> Optional[str]:
         """Tempo endpoint for charm tracing."""
-        return self.tempo_cluster.get_tracing_endpoint("otlp_http")
+        if endpoints := self.worker.cluster.get_tracing_receivers():
+            return endpoints.get("otlp_http", None)
 
     @property
     def ca_cert_path(self) -> Optional[str]:
         """CA certificate path for tls tracing."""
-        return TEMPO_CA_FILE if Path(TEMPO_CA_FILE).exists() else None
+        return CLIENT_CA_FILE
+
+    def pebble_layer(self, worker: Worker) -> Layer:
+        """Return a dictionary representing a Pebble layer.
+
+        Caller is responsible for checking whether the tempo role is valid before
+        calling this method.
+        """
+        role = "".join(worker.roles)
+        if role == "all":
+            role = "scalable-single-binary"
```

Comment:
Not sure, but should this be:
```suggestion
        role = "scalable-single-binary" if "all" in worker.roles else "".join(worker.roles)
```